src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1274,25 +1274,48 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           return glnx_throw (error, "No overrides for package '%s'", name_or_nevra);
         }
 
-      changed = TRUE;
-    }
-
-  if (self->override_replace_local_pkgs)
-    {
-      g_autoptr(GPtrArray) pkgs = NULL;
-      if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
-                                   cancellable, error))
-        return FALSE;
-
-      if (pkgs->len > 0)
+      if (self->override_replace_local_pkgs)
         {
-          g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_overrides (origin, (char**)pkgs->pdata,
-                                               RPMOSTREE_ORIGIN_OVERRIDE_REPLACE_LOCAL,
-                                               error))
+          g_autoptr(GPtrArray) pkgs = NULL;
+          if (!import_many_local_rpms (repo, self->override_replace_local_pkgs, &pkgs,
+                                       cancellable, error))
             return FALSE;
-          changed = TRUE;
-        }
+
+          for (char **it = (char**)pkgs->pdata; it && *it; it++)
+            {
+
+              const char *pkg = *it;
+              g_autofree char *name = NULL;
+              g_autofree char *sha256 = NULL;
+
+              if(!rpmostree_decompose_sha256_nevra(&pkg, &sha256, error))
+                return FALSE;
+
+              if(!rpmostree_decompose_nevra(pkg, &name, NULL, NULL, NULL, NULL, error))
+                return FALSE;
+
+              auto nevra = static_cast<const char *>(g_hash_table_lookup (name_to_nevra, name));
+
+              if (nevra != NULL)
+                {
+                  /* completely brush over the ridiculous corner-case of a
+                     pkgname that's also a nevra for another package */
+                  g_assert_not_reached ();
```

Comment:
Hmm, leftover from refactor? This will crash the process but having `nevra != NULL` is what we expect in the case of a match.

Reply:
Check out https://developer.gnome.org/glib/unstable/glib-Testing.html for documentation about those `g_assert_*` helpers.